test/plugins/alias-completion.plugin.bats

```diff
@@ -22,3 +22,10 @@ load ../../completion/available/git.completion
 
   assert_success
 }
+
+@test "alias-completion: See that having aliased rm command does not output unnecessary output" {
+  alias rm='rm -v'
```

Comment:
In order for the test to be useful, we need a way to *prove* that `command rm` gets invoked.
A quick idea:
* Path-munge a test-specific `rm` that does something like `mv $1 $1.rm` (would probably need more logic)
* Make the alias invoke something like `false` 
* Confirm that `file.rm` exists
If the alias gets called, then `file.rm` will not exist.

Reply:
It's probably easier to just make the alias to `mv` the file, and assert that it does not exist afterwards

Reply:
Well we want the test to confirm a few things:
1. The script section that contains `rm` *was* executed
2. The `rm` command *was* invoked
3. The `rm` alias was *not* invoked

Passing the test requires confirming some observable behavior that *only* exists when all of these are true.

The existing `refute_output` check for example, is poor because, although it can confirm if \#3 fails, it can't confirm that \#1 passes.

So by path-munging a custom `rm` script, you get the ability to create some type of observable outcome that proves it was invoked.  Additionally, in this case, having the alias invoke the path-munged command could muddy the test, unless it somehow affects the observed results.
 
NOTE: I actually don't know if trying to accurately test this behavior is worth the effort

Reply:
@davidpfarrell take a look at #1757, I think this PR addresses this problem